rama-http-backend/src/client/svc.rs

```diff
@@ -47,11 +47,12 @@ where
         // directly instead of here...
         let req = sanitize_client_req_header(&mut ctx, req)?;
 
-        let resp = match &self.0 {
+        let mut resp = match &self.0 {
             SendRequest::Http1(sender) => sender.lock().await.send_request(req).await,
             SendRequest::Http2(sender) => sender.lock().await.send_request(req).await,
         }?;
 
+        resp.extensions_mut().insert(ctx);
```

Comment:
Do we want to return ctx, similar to how it is done in other places, or inject it in request extensions? Injecting it in extensions is pretty easy and transparent to use

Reply:
Did not add e2e test for this as it is very small, and usage will be included in the dynamic certificate issuer e2e test.

Example of how to use this
```rust
inner_client.set_tls_config(ClientConfig {
    store_server_certificate_chain: true,
    ..Default::default()
});

...

let neg_tls_params = response
    .extensions()
    .get::<Context<()>>()
    .unwrap()
    .get::<NegotiatedTlsParameters>()
    .unwrap();

let certificates = neg_tls_params.peer_certificate_chain.clone().unwrap();
```

Reply:
Injecting context into the response feels dirty/wrong.

But to be fair, the fact that the context dies at the end of a chain did always botter me...

Still in spirit of current codebase this doesn't feel like the right solution. Instead one probably would use a layer to extract that information out before the context is being passed on to die.

It's unconvenient, that's true, but for now it's the only way to go I think.

Reply:
As discussed in private, this is part of a bigger problem that is out of scope for this PR. As such a new issue will be created where we will further discuss this problem and solutions. Once a solution is merged, I will also use update code here to use this new approach to expose inner layer stuff to the outside